### DISCUSSION TITLE
How to set default language?

### DISCUSSION THREAD

**Discussion by ChenYouRen** [primefaces/community#2707] [PrimeNG]
I use dataTable component and found it showed simplified Chinese text "无相关数据。" when it's empty.
![螢幕擷取畫面 2024-09-18 120306](https://github.com/user-attachments/assets/36f46ecf-19d4-4450-9600-dbbbf901ce0a)
How to set the default setting to traditional Chinese?

**Comment by qwadrox** [comment]
@ChenYouRen Hi. I don't entirely understand your question. 

You can change the locale of the components as so: https://primeng.org/configuration#locale

And here you can find the repository of the predefined locale files: https://github.com/primefaces/primelocale

You might need to create your own locale file. If yes, you can also create a pull request to the locale repo to share it 💯

**Reply by qwadrox** [reply]
Does this answer your question?

**Reply by ChenYouRen** [reply]
Thank you for your reply.
I use primefaces to write JSF. (Not NG, excuse me.)
It seem have a setting to set default locale (to show traditional Chinese message/text), but I don't know where and how to set.

**Reply by ChenYouRen** [reply]
Embarrassed...
The locale of message text setting is from web side. (Not from Primefaces (maybe...))
Use `<f:view locale="en-us">` to forced change language.
Also:
https://stackoverflow.com/questions/20176033/default-locale-does-not-seem-to-work-instead-the-supported-locale-is-being
https://stackoverflow.com/questions/4830588/localization-in-jsf-how-to-remember-selected-locale-per-session-instead-of-per
Thank you. 😅
